packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -826,98 +843,115 @@ class GraphiQLWithContext extends React.Component<
+                      <section
+                        className="graphiql-editor-tool"
+                        aria-label={
+                          this.state.activeSecondaryEditor === 'variable'
+                            ? 'Variables'
+                            : 'Headers'
+                        }>
+                        <VariableEditor
+                          onEdit={this.props.onEditVariables}
                           editorTheme={this.props.editorTheme}
-                          onEdit={this.props.onEditHeaders}
                           readOnly={this.props.readOnly}
-                          shouldPersistHeaders={this.props.shouldPersistHeaders}
+                          isHidden={
+                            this.state.activeSecondaryEditor !== 'variable'
+                          }
                           keyMap={this.props.keyMap}
                         />
-                      )}
-                    </section>
+                        {headerEditorEnabled && (
+                          <HeaderEditor
+                            isHidden={
+                              this.state.activeSecondaryEditor !== 'header'
+                            }
+                            editorTheme={this.props.editorTheme}
+                            onEdit={this.props.onEditHeaders}
+                            readOnly={this.props.readOnly}
+                            shouldPersistHeaders={
+                              this.props.shouldPersistHeaders
+                            }
+                            keyMap={this.props.keyMap}
+                          />
+                        )}
+                      </section>
+                    </div>
                   </div>
                 </div>
-              </div>
-              <div ref={this.props.editorResize.dragBarRef}>
-                <div className="graphiql-horizontal-drag-bar" />
-              </div>
-              <div ref={this.props.editorResize.secondRef}>
-                <div className="graphiql-response">
-                  {this.props.executionContext.isFetching ? (
-                    <div className="graphiql-spinner" />
-                  ) : null}
-                  <ResponseEditor
-                    value={this.props.response}
-                    editorTheme={this.props.editorTheme}
-                    ResponseTooltip={this.props.ResultsTooltip}
-                    keyMap={this.props.keyMap}
-                  />
-                  {footer}
+                <div ref={this.props.editorResize.dragBarRef}>
+                  <div className="graphiql-horizontal-drag-bar" />
+                </div>
+                <div ref={this.props.editorResize.secondRef}>
+                  <div className="graphiql-response">
+                    {this.props.executionContext.isFetching ? (
+                      <div className="graphiql-spinner" />
+                    ) : null}
```

Comment:
```suggestion
                    {this.props.executionContext.isFetching && (
                      <div className="graphiql-spinner" />
                    )}
```

Reply:
This seems to me like a clash of personal preferences, because I like ternaries better 😄 (for the simple reason that it avoids the pitfall of including `0` in the markup when the variable you base the assertion on is a number)

I'm open to being more opinionated here (no matter which one we choose), but if we think that this brings us value we should imho look into validating this using `eslint` and uniformly adapt the whole codebase. So I'm just leaving this as is for now. ❤️